flink-autoscaler/src/main/java/org/apache/flink/autoscaler/JobAutoScalerImpl.java

```diff
@@ -160,14 +160,23 @@ private void runScalingLogic(Context ctx, AutoscalerFlinkMetrics autoscalerMetri
         var collectedMetrics = metricsCollector.updateMetrics(ctx, stateStore);
         var jobTopology = collectedMetrics.getJobTopology();
 
+        var now = clock.instant();
```

Comment:
Hi @afedulov ,

We are using `now` as the `endTime` of ScalingRecord. The `kubernetes.operator.reconcile.interval` is 1 min by default and  `scaling` is called in each reconcile. 

So, the restartTime will be greater than `1 min` even if the scaling is very quick.

I'm not sure should we get the endTime from the `JobUpdateTs`? Currently, `stabilization.interval` checks the `JobUpdateTs`.

Looking forward to your feedback, thanks~

Reply:
Reconciliation is triggered at least once per minute, but actual reconciliation time is more like 10-20 seconds. This is due cluster events also triggering the reconciliation loop. However, I think you are right, we should use the job update time and not the current time. Let's wait what Alex says.

Reply:
@mxm I just checked and unfortunately our assumption does not seem to hold true with that regard. The loop is not triggered when the job state changes to `RUNNING`. I tried setting a higher reconciliation interval and it directly is reflected in the recorded durations.

Reply:
>I just checked and, unfortunately, our assumption does not seem to hold true with that regard. The loop is not triggered when the job state changes to RUNNING. I tried setting a higher reconciliation interval and it directly is reflected in the recorded durations.

There is no strict guarantee how often we are being called, but in my tests I saw 10-20 seconds due to cluster events arriving. In any case, the main issue here is not the reconciliation interval but that we are using the current instant instead of the job update time. Let's address this as proposed by Rui and we should be good to go. This feature might then be robust enough to be enabled out of the box.